src/Adapter/InMemoryAdapter.php

```diff
@@ -135,18 +136,22 @@ public function loadReverse(
 
     public function loadEvents(
         StreamName $streamName,
-        array $metadata = [],
         int $fromNumber = 0,
-        int $count = null
+        int $count = null,
+        MetadataMatcher $metadataMatcher = null
     ): Iterator {
         if (! isset($this->streams[$streamName->toString()])) {
```

Comment:
I would prefer a hasStream method for readability. Maybe with a StreamCollection.